### ISSUE TITLE
vcf import should update existing records, not duplicate them

### ISSUE THREAD

**Issue by cboltz** [owncloud/contacts#588]
### Steps to reproduce
1. Login to the ownCloud web interface
2. go to the contacts app
3. import a vcf file (exported from somewhere else)
4. (optional) slightly edit the vcf file, for example change a phone number
5. import that vcf file again

### Expected behaviour
The address book entry created after step 3 (hint: same `UID:`) should be updated (if the vcf file was modified)

### Actual behaviour
Two adress book entries with same or (if the vcf was modified) slightly different content. 

This makes staying in sync with an external address list very hard - instead of simply importing the updated vcf file, I have to apply changes manually.

### Server configuration
openSUSE Leap 42.2, Apache, mariadb, PHP 7.0.7

**ownCloud version:** 9.1.4

**Contacts version:** 1.5.1

**Updated from an older ownCloud or fresh install:** fresh install

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

(CC @ccboltz)

**Comment by Julian1998** [comment]
A fix for that is not that easy. Hopefully we can implement it next version.
You can't simply update duplicates. So in case of duplicates the user has to decide if he either wants to override, merge or ignore that contact.

**Comment by cboltz** [comment]
I'm a bit confused because you closed this ticket. Do you have another place to track the hard-to-implement things? If so, is there a way to get notifications when you work on this?

BTW: For my usecase, overwriting the existing contact (VCF file always wins) would be perfect, but it probably makes sense to let the user decide on that. As a first step, not creating duplicate address book entries if the data in ownCloud and the VCF file is exactly the same would already be very helpful.

**Comment by Julian1998** [comment]
@cboltz https://github.com/owncloud/contacts/issues/632

I summed the possible events when importing a vfc file up in a new ticket

If you want to I can tell you what you have to change for your usecase (It's pretty simple) 
I don't think that I can implement that feature the next few months

**Comment by cboltz** [comment]
Thanks for the pointer to the new ticket!

And yes, some hints about what I need to change to avoid duplicates would be welcome ;-)

**Comment by rubo77** [comment]
Any hints?

I just had a big problem: in davdroid I deleted all contracts, because there were just 200 from my phone contacts and immediately copied all 800 contracts from my phone to the davdroid account. 

Then all 800 were synced to Nextcloud but the initial deletion was ignored, so I end up with 1000 contacts in Nextcloud. 200 duplicates!

This issue should be reopened

**Comment by Julian1998** [comment]
@rubo77 There should be a different ticket for this issue because it's way more complex... But it's on our list ;)
